sys/include/riot_hdr.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2018 Kaspar Schleiser <kaspar@schleiser.de>
+ *                    Inria
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_riot_hdr RIOT header helpers and tools
+ * @ingroup     sys
+ * @{
+ *
+ * The header contains
+ *
+ * - "RIOT" as magic number
+ * - the application version
+ * - the address where to find the RIOT firmware
+ * - the checksum of the three previous fields
+ *
+ * @file
+ * @brief       RIOT firmware header and tools
+ *
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ * @author      Francisco Acosta <francisco.acosta@inria.fr>
+ *
+ * @}
+ */
+
+#ifndef RIOT_HDR_H
+#define RIOT_HDR_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+
+/**
+ * @brief  Magic number for riot_hdr
+ *
+ */
+#define RIOT_HDR_MAGIC         0x544f4952 /* "RIOT" */
+
+/**
+ * @brief Structure to store image header - All members are little endian
+ * @{
+ */
+typedef struct {
+    uint32_t magic_number;         /**< header magic_number (always "RIOT")    */
```

Comment:
+1 I asked to precise the endianess of all fields.

Reply:
I really don't get what do you mean by this. Should I put an attribute to each field to express the endianness? Otherwise I don't know how to make explicit the endianness of a type or variable in C.

Reply:
Sorry I missed it was written one line before the github shown section.